lti_consumer/lti_xblock.py

```diff
@@ -1405,6 +1414,59 @@ def _get_lti_launch_url(self, consumer):
 
         return launch_url
 
+    def get_1p3_launch_data(self):
+        """
+        Return a dictionary containing necessary data for an LTI 1.3 launch.
+        """
+        context_information = self.get_context_launch_information()
+
+        user_role = self.role
+        external_user_id = self.external_user_id
+
+        resource_link_information = {
+            "id": str(self.location),  # pylint: disable=no-member
+        }
+
+        launch_presentation_information = {
+            "document_target": "iframe",
+        }
+
+        # Runtime import since this will only run in the
+        # Open edX LMS/Studio environments.
+        # pylint: disable=import-outside-toplevel
+        from lti_consumer.api import _get_lti_config
+        lti_config = _get_lti_config(block=self)
+
+        return {
+            "config_id": str(lti_config.config_id),
+            "message_type": "LtiResourceLinkRequest",
+            "user_id": external_user_id,
+            "user_role": user_role,
+            "context": context_information,
+            "launch_presentation": launch_presentation_information,
+            "resource_link": resource_link_information,
+        }
```

Comment:
we don't really need anything other than a dictionary but if we made this a python object it would be more obvious when you put in weird stuff. And it would be a place to put the documentation in the big comment above that describes the launch structure.

Reply:
+1 for making this an object. We could even use that class to hide away how we encode/decode/validate this which might help keep the consumer class cleaner.